server/src/repositories/search.repository.ts

```diff
@@ -73,10 +73,17 @@ export class SearchRepository implements ISearchRepository {
   async searchMetadata(pagination: SearchPaginationOptions, options: AssetSearchOptions): Paginated<AssetEntity> {
     let builder = this.assetRepository.createQueryBuilder('asset');
     builder = searchAssetBuilder(builder, options);
-
+    let mode = PaginationMode.SKIP_TAKE;
     builder.orderBy('asset.fileCreatedAt', options.orderDirection ?? 'DESC');
+
+    if (options.random) {
+      mode = PaginationMode.LIMIT_OFFSET;
```

Comment:
I don't think limit/offset is safe here because there are possible joins with one-to-many or many-to-many relations.

Reply:
A thought: you could make a separate `/search/random` endpoint that takes the same options as the smart search endpoint but without the query field.

Reply:
I see. The other pagination throws an error I did not know how to fix. If you have time can you take a look?

Reply:
Nevermind, I think I figured it out.